ql/src/java/org/apache/hadoop/hive/ql/parse/BaseSemanticAnalyzer.java

```diff
@@ -1926,6 +1960,19 @@ protected Database getDatabase(String dbName, boolean throwException) throws Sem
     return database;
   }
 
+  protected Database getDatabase(String catalogName, String dbName, boolean throwException) throws SemanticException {
```

Comment:
try reuse code. you could call new method within the old one without catalogName